src/screens/ImportAccountSelectLedgerApp/index.tsx

```diff
@@ -57,10 +59,22 @@ const ImportAccountSelectLedgerApp: React.FC<NavProps> = ({ navigation }) => {
       });
 
       if (account !== undefined) {
-        onSuccess({ account, chain: selectedChain! });
+        if (storeAccount) {
+          saveAccount(account);
+        } else {
+          onSuccess({ account, chain: selectedChain! });
+        }
```

Comment:
Why is this an `if-else`? Shouldn't it be only an `if` calling the `saveAccount`, and then `onSuccess` being called in any case?